### DISCUSSION TITLE
Rust frontend-backend communication without JSON in-between possible?

### DISCUSSION THREAD

**Discussion by JakubKoralewski** [tauri-apps/tauri#1268] [General]
Hello! I'm currently reading the docs to try to understand the project and I have a question:

https://tauri.studio/en/docs/guides/command mentions:

> Tauri by default serializes every message to a JSON string, so you can use serde_json on the backend to deserialize it.

But if I write my frontend in Rust and use Rust for handling the commands the in between step of converting a Rust struct into JSON and back again seems redundant. Could something like https://github.com/servo/bincode be used for the message passing? Is this too niche a feature?

Maybe I'm misunderstanding how message/command passing works, and even though I'd use Rust in the frontend, it would generate a JS call anyway?

Thanks

**Comment by nklayman** [comment]
Currently this isn't supported, but it's definitely something we're planning to add in the future.

**Reply by nklayman** [reply]
You should go ahead and open a feature request issue for this so we can track it better, as it might get lost here :)

**Reply by JakubKoralewski** [reply]
Will do!

**Reply by tensor-programming** [reply]
I wonder if it even works this way.  The rust frontend would generate WASM.  And the WASM would be injected from the generated HTML/CSS/JS etc.  The Wasm would be static and it wouldn't need to be serialized.   I don't think it would even go through the tauri event manager. Though if thats the case, perhaps its a security concern.

**Reply by lucasfernog** [reply]
and if you want to use the tauri API directly from Rust code, you can use the `tauri-api` crate, re-exported as `tauri::api`.
